crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1189 @@
+                }
+                Expr::Await(await_expr)
+            }
+            Expr::This(this_expr) => {
+                if !self.config.allow_top_level_this && !self.enter_fn {
+                    return *undefined(DUMMY_SP);
+                }
+                Expr::This(this_expr)
+            }
+            _ => expr.fold_children_with(self),
+        }
+    }
+
+    fn fold_fn_decl(&mut self, fn_decl: FnDecl) -> FnDecl {
+        let is_async = fn_decl.function.is_async;
+        if is_async {
+            self.enter_async_fn = true;
+        }
+        self.enter_fn = true;
+        let fold_fn_expr = fn_decl.fold_children_with(self);
+        if is_async {
+            self.enter_async_fn = false;
+        }
+        self.enter_fn = false;
+        fold_fn_expr
+    }
+
+    fn fold_prop(&mut self, prop: Prop) -> Prop {
+        match prop {
+            Prop::Shorthand(shorthand) => Prop::KeyValue(KeyValueProp {
+                key: PropName::Ident(shorthand.clone()),
+                value: Box::new(self.fold_module_name_ident(shorthand)),
+            }),
+            Prop::KeyValue(key_value_prop) => Prop::KeyValue(KeyValueProp {
+                key: key_value_prop.key,
+                value: match *key_value_prop.value {
+                    Expr::Ident(ident) => Box::new(self.fold_module_name_ident(ident)),
+                    _ => key_value_prop.value,
+                },
+            }),
+            _ => prop,
+        }
+    }
+
+    fn fold_module(&mut self, module: Module) -> Module {
+        let mut before_body_stmts: Vec<Stmt> = vec![];
+        let mut execute_stmts = vec![];
+
+        // collect top level fn decl
+        for item in &module.body {
+            if let ModuleItem::Stmt(Stmt::Decl(Decl::Fn(fn_decl))) = item {
+                self.root_fn_decl_idents.push(fn_decl.ident.clone());
+            }
+        }
+
+        for item in module.body {
+            match item {
+                ModuleItem::ModuleDecl(decl) => match decl {
+                    ModuleDecl::Import(import) => {
+                        let src = match &self.resolver {
+                            Some((resolver, base)) => resolver
+                                .resolve_import(base, &import.src.value)
+                                .with_context(|| {
+                                    format!("failed to resolve import `{}`", import.src.value)
+                                })
+                                .unwrap(),
+                            None => import.src.value,
+                        };
+
+                        let source_alias = local_name_for_src(&src);
+
+                        let mut setter_fn_stmts = vec![];
+
+                        for specifier in import.specifiers {
+                            match specifier {
+                                ImportSpecifier::Default(specifier) => {
+                                    self.import_ident_names.push(specifier.local.sym.clone());
```

Comment:
Seems like this will not work with identifier management system of swc.

You can read about it at https://rustdoc.swc.rs/swc_ecma_ast/struct.Ident.html